### PR TITLE
[dynamo] add meta fn for aten.kthvalue.default

### PR DESCRIPTION
**Pull request by ColinPeppler**
I saw
```
torch._dynamo.exc.Unsupported: unsupported operator: aten.kthvalue.default
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130562

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -5772,10 +5772,6 @@ def scalar_tensor(s, dtype=None, layout=None, device=None, pin_memory=None):
 def topk_meta(self, k, dim=-1, largest=True, sorted=True):
     # From aten/src/ATen/native/Sorting.cpp
     dim = maybe_wrap_dim(dim, self.dim(), wrap_scalar=True)
-    torch._check(
-        k >= 0 and k <= (self.size(dim) if self.dim() > 0 else 1),
-        lambda: "selected index k out of range",
-    )
     sliceSize = 1 if self.dim() == 0 else self.size(dim)
     torch._check(k >= 0 and k <= sliceSize, lambda: "k not in range for dimension")
```
Comment (ColinPeppler) [inline torch/_meta_registrations.py:5776]: Found out these two `torch._checks` check the same thing. So, I removed the first one :)

**Review by jingsh — APPROVED** [review]
lgtm

**Comment by jingsh** [comment]
seems some CI failed, related?

**Comment by aakhundov** [comment]
> seems some CI failed, related?

Yeah, CI failures look legit.

**Comment by ColinPeppler** [comment]
Also seeing this, but not sure how to proceed on this exactly. I have no issue with `topk.values`.

```
NotImplementedError: aten::kthvalue.values: attempted to run this operator with Meta tensors, but there was no fake impl or Meta kernel registered. You may have run into this message while using an operator with PT2 compilation APIs (torch.compile/torch.export); in order to use this operator with those APIs you'll need to add a fake impl. Please see the following for next steps:  https://pytorch.org/tutorials/advanced/custom_ops_landing_page.html
```

**Comment by ColinPeppler** [comment]
Yeah, CI failures were legit but all but one were unexpected success ;)

File: test/test_proxy_tensor.py [hunk 2]
```diff
@@ -1999,7 +1999,6 @@ def f(t):
     xfail('geqrf', ''),  # aten.geqrf.default - couldn't find symbolic meta function/decomposition
     xfail('histogram', ''),  # Could not run 'aten::histogram.bin_ct' with arguments from the 'Meta' backend. This c...
     xfail('histogramdd', ''),  # aten._histogramdd_bin_edges.default - couldn't find symbolic meta function/decomposition
-    xfail('kthvalue', ''),  # aten.kthvalue.default - couldn't find symbolic meta function/decomposition
```
Comment (zou3519) [inline test/test_proxy_tensor.py:2002]: Add `kthvalue` to out_symbolic_tensor_failures. You added a meta for one of the two kthvalue operators but not the other one. This xfail was blocking both of those from showing up in CI

File: torch/_meta_registrations.py [hunk 3]
```diff
@@ -5785,6 +5781,22 @@ def topk_meta(self, k, dim=-1, largest=True, sorted=True):
     return self.new_empty(topKSize), self.new_empty(topKSize, dtype=torch.int64)
 
 
+@register_meta([aten.kthvalue.default, aten.kthvalue.values])
+@out_wrapper("values", "indices")
```
Comment (ColinPeppler) [inline torch/_meta_registrations.py:5785]: Adding this helped

**Review by zou3519 — APPROVED** [review]

File: torch/_meta_registrations.py (hunk 3 above)
Comment (zou3519) [reply torch/_meta_registrations.py:5785]: nice

**Review by jingsh — APPROVED** [review]

**Pull request closed without merging**